app/js/instances/instanceDetailController.js

```diff
@@ -21,32 +21,51 @@
 
 var _ = require('lodash');
 
-function instanceDetailController($scope, $routeParams, dockerService, instanceModel, instanceService) {
+function instanceDetailController($scope, $routeParams, $timeout, flux, dockerService, instanceModel, instanceService) {
     'use strict';
 
     // init
     $scope.containers = [];
     $scope.images = [];
+    $scope.loadingImages = {};
     $scope.instance = {alias: $routeParams.host};
-    $scope.allImages = false;
+    $scope.allImages = instanceModel.getShowAllImages();
     $scope.allContainers = false;
 
     dockerService.d('containers.list', {
-        host: $scope.instance.alias
+        host: $scope.instance.alias,
+        query: {
+            all: $scope.allContainers
+        }
     });
 
     dockerService.d('images.list', {
-        host: $scope.instance.alias
+        host: $scope.instance.alias,
+        query: {
+            all: $scope.allImages
+        }
     });
 
     // State event listeners
     $scope.$listenTo(instanceModel, function () {
         $scope.containers = instanceModel.getContainers();
         $scope.images = instanceModel.getImages();
+
+        var loadingImages = instanceModel.getLoadingImages();
+        _.forEach(loadingImages, function(progress, name) {
+            $scope.loadingImages[name] = progress;
+        });
+        $scope.loadingImages = _.pick($scope.loadingImages, _.keys(loadingImages));
+
+        $timeout(function() {
```

Comment:
Nah it was doing it before I set that up. I also iteratively update the object so the loading bar's css transitions are smooth.  Otherwise if I overwrite the whole object the loading bar doesn't do the transitions correctly and is all jittery.